src/magazine.py

```diff
@@ -40,39 +48,62 @@ def download_magazine(self, id):
             done = False
             while done is False:
                 status, done = downloader.next_chunk()
-
+                
         except HttpError as error:
             print(f"An error occurred: {error}")
             file = None
+            
         if file is not None:
-            payload = json.dumps(
-                {
-                    "bucket": UPLOAD_BUCKET,
-                    "image": "data:application/pdf;base64,"
-                    + base64.b64encode(file.getvalue()).decode("utf-8"),
-                }
-            )
-            return requests.post(
-                "https://upload.cornellappdev.com/upload/",
-                payload,
-            ).text
+            pdf = pdfium.PdfDocument(file)
+            first_page = pdf[0]
+            pil_image = first_page.render().to_pil().resize((150,220))
+
+            imageBytes = io.BytesIO() #create bytes stream to hold image
+            pil_image.save(imageBytes, format='png')
+            
+            image_payload = json.dumps({
+                "bucket": "volume",
+                "image": "data:image/png;base64,"
+                + base64.b64encode(imageBytes.getvalue()).decode("utf-8")
+            })
```

Comment:
What would be best for me is if you slightly changed your refactoring by splitting up download_content into two functions: One that returns the Bytes_IO object of the content, and another that takes in a Bytes_IO and posts it to digital ocean, returning the same thing that download_magazine currently does. 

For this second function, it could either be specific to the content-type (pdf, image) or could take in a certain content type as a parameter. Attached is an updated utils.py with how you might do it with separate content-types for pdf and image.

[utils_new.txt](https://github.com/cuappdev/volume-microservice/files/11314435/utils_new.txt)

Reply:
Note that with this approach you will have to import pdfium2 as pdfium in the file as I did in utils_new.txt- This will also require adding pdfium2 and adding:
Pillow==9.5.0
pypdfium2==4.4.0

to the requirements.txt file.